lib/tools/getYouTubeChannelInfo.ts

```diff
@@ -0,0 +1,118 @@
+    "or clear authentication instructions if tokens are missing/expired. " +
+    "IMPORTANT: This tool requires the artist_id parameter. If you don't know the artist_id, ask the user or use the current artist's artist_id.",
+  parameters: schema,
+  execute: async ({ artist_id }): Promise<YouTubeChannelInfoResult> => {
+    // Early validation of artist_id
+    if (!artist_id || artist_id.trim() === '') {
+      const missingParamError = YouTubeErrorBuilder.createToolError(
+        "No artist_id provided to YouTube tool. The LLM must pass the artist_id parameter. Please ensure you're passing the current artist's artist_id."
+      );
+      return missingParamError;
+    }
+    
+    try {
+      // Validate YouTube tokens (internal authentication check)
+      const tokenValidation = await validateYouTubeTokens(artist_id);
+      
+      if (!tokenValidation.success) {
+        // Return authentication error with clear instructions
+        const authError = YouTubeErrorBuilder.createToolError(
+          `YouTube authentication required for this account. ${tokenValidation.error!.message} Please authenticate by connecting your YouTube account.`
+        );
+        return authError;
+      }
+
+      // Fetch comprehensive channel information with branding
+      const channelResult = await fetchYouTubeChannelInfo(tokenValidation.tokens!, true);
+      
+      if (!channelResult.success) {
+        const fetchError = YouTubeErrorBuilder.createToolError(channelResult.error!.message);
+        return fetchError;
+      }
+
+      const channel = channelResult.channelData!;
+      
+      const returnResult = YouTubeErrorBuilder.createToolSuccess("YouTube channel information retrieved successfully", {
+        channelInfo: {
+          // Basic channel information
+          id: channel.id,
+          title: channel.title,
+          description: channel.description,
+          customUrl: channel.customUrl,
+          country: channel.country,
+          publishedAt: channel.publishedAt,
+          
+          // Channel thumbnails
+          thumbnails: channel.thumbnails,
+          
+          // Channel statistics
+          statistics: {
+            subscriberCount: channel.statistics.subscriberCount,
+            videoCount: channel.statistics.videoCount,
+            viewCount: channel.statistics.viewCount,
+            hiddenSubscriberCount: channel.statistics.hiddenSubscriberCount || false,
+          },
+          
+          // Channel branding
+          branding: {
+            keywords: channel.branding?.keywords || null,
+            defaultLanguage: channel.branding?.defaultLanguage || null,
+          },
+          
+          // Authentication metadata
+          authentication: {
+            tokenCreatedAt: tokenValidation.tokens!.created_at,
+            tokenExpiresAt: tokenValidation.tokens!.expires_at,
+          },
+        }
+      });
```

Comment:
why are you destructuring the response?
- code would be more efficient to simply return `{...channel, ... tokenValidation}` right?